defs/channel.d.ts

```diff
@@ -0,0 +1,228 @@
+  /**
+   * The current viewer count if the channel is live.
+   */
+  viewersCurrent: number;
+  /**
+   * Total amount of followers to the channel.
+   */
+  numFollowers: number;
+  /**
+   * The description for the channel.
+   */
+  description: string;
+  /**
+   * The typeId for the game that that is being played / streamed.
+   */
+  typeId: number;
+  /**
+   * Is the channel using an interactive game / controls.
+   */
+  interactive: boolean;
+  /**
+   * The Id of the tetris game.
+   */
+  tetrisGameId: number;
+  /**
+   * If this is not 0 the channel is streaming using FTL.
+   */
+  ftl: number;
+  /**
+   * Has the channel got a VOD (Previous stream video)
+   */
+  hasVod: boolean;
+  /**
+   * TODO: Find out what this is for.
+   */
+  languageId: number;
+  /**
+   * The Id of the cover the channel has set.
+   */
+  coverId: number;
+  /**
+   * TODO: Find out what this is for.
+   */
+  thumbnailId: number;
+  /**
+   * TODO: Find out what this is for.
+   */
+  badgeId: number;
+  /**
+   * The channel Id of the channel being hosted.
+   */
+  hosteeId: number;
+  /**
+   * When the channel was created.
+   */
+  createdAt: string;
+  /**
+   * When the channel was last updated.
+   */
+  updatedAt: string;
+  /**
+   * When the channel was deleted.
+   */
+  deletedAt: string;
+  /**
+   * The background cover for the channel.
+   */
+  cover: {
+    /**
+     * Meta data about the cover.
+     */
+    meta: {
+      /**
+       * The URL to the small version of the cover.
+       */
+      small: string;
+    }
+    /**
+     * The Id for the cover.
+     */
+    id: number;
+    /**
+     * The type of the cover.
+     */
+    type: string;
+    /**
+     * TODO: Find out what this is for.
+     */
+    relid: number;
+    /**
+     * The url the full version of the cover.
+     */
+    url: string;
+    /**
+     * Remote path for the cover (Used internally)
+     */
+    remotePath: string;
+    /**
+     * When the cover was created.
+     */
+    createdAt: string;
+    /**
+     * When the cover was updated.
+     */
+    updatedAt: string;
+  };
+  /**
+   * TODO: Find out what this is for.
+   */
+  badge: string;
+  /**
+   * Data about the "game" type being streamed.
+   */
+  type: {
+    /**
+     * The type Id.
+     */
+    id: number;
+    /**
+     * The name of the type.
+     */
+    name: string;
+    /**
+     * The category for the type.
+     */
+    parent: string;
+    /**
+     * The description for the type.
+     */
+    description: string;
+    /**
+     * The source for when the type came from. I.E. player.me
+     */
+    source: string;
+    /**
+     * Total amount of viewers watching this type.
+     */
+    viewersCurrent: number;
+    /**
+     * The cover Url for this type.
+     */
+    coverUrl: string;
+    /**
+     * The amount of channels online using this type.
+     */
+    online: number;
+  };
+}
+
+export interface BeamChannelUser<T> extends BeamChannel {
```

Comment:
`T` is just extending a definition which is passed over. See [here](https://github.com/WatchBeam/beam-client-node/pull/20/files#diff-266fc8f60b3b9dc6a4d3d1be2ebd0a79R11) and [here](https://github.com/WatchBeam/beam-client-node/pull/20/files#diff-266fc8f60b3b9dc6a4d3d1be2ebd0a79R15) that I have two definitions for the user which one has more data because you are authed. Along with this makes updating the definitions easier when more things are added; on which the code is neater and more manageable.

The method is not really needed but it was done for looks and management, easy importing when you want the definitions at application level rather that module.